tests/Fixtures/TestBundle/HttpCache/TagCollectorCustom.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\HttpCache;
+
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Serializer\TagCollectorInterface;
+use ApiPlatform\Tests\Fixtures\TestBundle\Entity\RelationEmbedder;
+
+/**
+ * Collects cache tags during normalization.
+ *
+ * @author Urban Suppiger <urban@suppiger.net>
+ */
+class TagCollectorCustom implements TagCollectorInterface
+{
+    public const IRI_RELATION_DELIMITER = '#';
+
+    public function collect(array $context = []): void
+    {
+        $iri = $context['iri'];
```

Comment:
IRI and object can be null here no? (we probably don't care as it's a test fixture) I'd probably add a comment for readers

Reply:
Yeah, `iri` and `object` could potentially both be null. I think this should be clearer now with the interface signature documentation.

In this specific implementation (`TagCollectorCustom`), I'm checking further below for this (e.g. `isset($iri)`).